haystack_experimental/components/retrievers/multi_query_keyword_retriever.py

```diff
@@ -0,0 +1,157 @@
+# SPDX-FileCopyrightText: 2022-present deepset GmbH <info@deepset.ai>
+#
+# SPDX-License-Identifier: Apache-2.0
+
+from concurrent.futures import ThreadPoolExecutor
+from typing import Any, List, Optional
+
+from haystack import Document, component, default_from_dict, default_to_dict
+from haystack.core.serialization import component_to_dict
+from haystack.utils.deserialization import deserialize_component_inplace
+
+from haystack_experimental.components.retrievers.types import KeywordRetriever
+
+
+@component
+class MultiQueryKeywordRetriever:
```

Comment:
I think we should also update this name to something like `MultiQueryTextRetriever`. For example, we have a retriever called `MongoDBAtlasFullTextRetriever` from our mongodb integration which follows the protocol but isn't exactly a keyword retriever.